segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -222,10 +221,166 @@ def generate_test_label_map(size_img=(1024, 1024), num_A=100, num_B=100, num_C=1
+    """
+    This function generates a test segmentation mask with each separate cell labeled separately.
+
+    Args:
+        size_img (tuple): the dimensions of the image we wish to generate
+        cell_radius (int): the radius of each cell
+
+    Returns:
+        sample_segmentation_mask (numpy): an array of dimensions size_img with two separate labeled
+            cells that border each other
+    """
+
+    # define the segmentation mask
+    sample_segmentation_mask = np.zeros(size_img)
+
+    # define the centers of the cells, need to subtract 1 from center 2 because of how
+    # the circle function in skimage.draw works
+    center_1 = (size_img[0] // 2, size_img[0] // 2)
+    center_2 = (size_img[0] // 2, size_img[0] // 2 + cell_radius * 2 - 1)
+
+    # generate the coordinates of each nuclear disk
+    cell_region_1_x, cell_region_1_y = circle(center_1[0], center_1[1], cell_radius, shape=size_img)
+    cell_region_2_x, cell_region_2_y = circle(center_2[0], center_2[1], cell_radius, shape=size_img)
+
+    # now assign the respective cells value according to their label
+    sample_segmentation_mask[cell_region_1_x, cell_region_1_y] = 1
+    sample_segmentation_mask[cell_region_2_x, cell_region_2_y] = 2
+
+    # we should define this dictionary to make it easy to index into the centers of each cell
+    # once we have to generate nuclear and membrane-level signal
+    # may need to change this to a different, immutable datatype
+    cell_centers = {1: center_1, 2: center_2}
+
+    return sample_segmentation_mask, cell_centers
+
+
+def generate_two_cell_test_nuclear_signal(segmentation_mask, cell_centers,
+                                          size_img=(1024, 1024), nuc_cell_ids=[1],
+                                          nuc_radius=3):
+    """
+    This function generates nuclear signal for the provided cells
+
+    Args:
+        segmentation_mask (numpy): an array which contains the labeled cell regions
+        cell_centers (dict): a dictionary which contains the centers associated with each cell region
+        size_img (tuple): the dimensions of the image we wish to generate
+        nuc_cell_ids (list): a list of cells we wish to generate nuclear signal for, if None assume just cell 1
+        nuc_radius (int): the radius of the nucleus of each cell
+
+    Returns:
+        sample_nuclear_signal (numpy): an array of equal dimensions to segmentation_mask
+            which have nuclear signal generated for the provided cell ids
+    """
+
+    # define the nuclear signal array
+    sample_nuclear_signal = np.zeros(segmentation_mask.shape)
+
+    for cell in nuc_cell_ids:
+        center = cell_centers[cell]
+
+        # generate the nuclear region in the middle of the cell with the same cell center
+        # and set signal to a uniform value
+        nuc_region_x, nuc_region_y = circle(center[0], center[1], nuc_radius, shape=size_img)
+        sample_nuclear_signal[nuc_region_x, nuc_region_y] = 1
```

Comment:
I find it's useful for debugging purposes if the signal isn't exactly equal to the area of the cell. Can change to 2 or 10 or whatever you want, just not 1.